Update CSC RU segment builder

nvoytish opened the pull request:
The RU builder thresholds are made looser on second and third iterations of the builder in order to increase the efficiency of the segment reconstruction for cases with "soft" muons, when the trajectory bends on a scale of a CSC detector. Related presentation can be found here https://indico.cern.ch/event/784929/contributions/3265299/attachments/1777781/2890942/19_01_11-Voytishin_Palichik-CSCSegBuilder_proposal_RECO_meeting.pdf

nvoytish committed:
Update RU builder

The RU builder thresholds are made looser on second and third iterations of the builder in order to increase the efficiency of the segment reconstruction for cases with "soft" muons, when the trajectory bends on a scale of a CSC detector. Related presentation can be found here https://indico.cern.ch/event/784929/contributions/3265299/attachments/1777781/2890942/19_01_11-Voytishin_Palichik-CSCSegBuilder_proposal_RECO_meeting.pdf

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25684/8039

- This PR adds an extra 20KB to repository

cmsbuild commented:
A new Pull Request was created by  @nvoytish  for master.

It involves the following packages:

RecoLocalMuon/CSCSegment


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@ptcox, @bellan, @abbiendi, @jhgoh this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

nvoytish committed:
Update CSCSegAlgoRU.cc

cmsbuild commented:
The code-checks are being triggered in jenkins.

nvoytish commented:
unused variable omitted

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25684/8041

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #25684 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0dca94e66ba9f1b36f9b815d9e95b9d81a1409f2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/32651/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/32651/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 8067 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3097440
 - DQMHistoTests: Total failures: 1554
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3095689
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

slava77 commented:
@nvoytish 
during the RECO meeting I understood that there was an issue with the efficiency loss only in ME11.
I see that the reco is modified for all stations.
This leads to a fairly significant increase in the number of reconstructed fake muons downstream.

I used a flat-pt QCD sample (muons are dominated by fakes) as a check and I see an increase of 19% in the number of PF muons
![all_sign1063vsorig_qcd13tevflatpt15s3000pu2018wf11071p0c_log10recopfcandidates_particleflow__reco_obj_pt51](https://user-images.githubusercontent.com/4676718/51716697-b1614780-1ff2-11e9-961f-6dc84b69365d.png)

![all_sign1063vsorig_qcd13tevflatpt15s3000pu2018wf11071p0c_recomuonsiminfoedmvaluemap_muonsimclassifier__reco_obj_values__extendedclass](https://user-images.githubusercontent.com/4676718/51717027-12d5e600-1ff4-11e9-8d1f-52cf040b6fc7.png)
https://github.com/cms-sw/cmssw/blob/master/DataFormats/MuonReco/interface/MuonSimInfo.h
Most of the increase is in "unmatched" (fake, extendedclass 0), punchthrough (1) and secondary muon (class 2), ghost (effectively a fake, class -1). It's quite unclear if any of these are useful.

similar, in ZMM sample with PU
![all_sign1063vsorig_zmm13tevpu2018wf11042p0c_recomuonsiminfoedmvaluemap_muonsimclassifier__reco_obj_values__extendedclass](https://user-images.githubusercontent.com/4676718/51717538-110d2200-1ff6-11e9-9a02-3bba5725d632.png)
There is a similarly significant increase in "fakes", while there is no visible increase in muons from Z (class 10) from their initial count of about 4000 in this test.

most of the added muons will not pass some tighter ID and will likely not affect most of the analyses.
However, those looking at soft muons may be affected.
In the QCD sample:
<img width="738" alt="wf11071_soft2_mass" src="https://user-images.githubusercontent.com/4676718/51718899-2f761c00-1ffc-11e9-9b64-36a7a378d565.png">
<img width="739" alt="wf11071_tight2_mass" src="https://user-images.githubusercontent.com/4676718/51718949-6cdaa980-1ffc-11e9-97f7-63277cececde.png">



On a global scale, it's quite unclear if a recovery of 0.5% of efficiency of good muons visible in ME11 (based on the slides in the reco meeting) is worth an increase of fake soft/pf muons by 20%.
I think that some cost-benefit analysis, impact minimization, and a preliminary OK from the muon POG is needed before I can sign off on this.
@drkovalskyi @folguera @bmahakud

nvoytish commented:
@slava77 
in ME11 we saw the biggest inefficiency in comparison with other stations, that's why we focused on ME11 (in presentation and while developing the solution for the improvement). But the improvement is applied for all stations. 
The plots you show are not surprising for me. When we developed the RU builder we were using MC samples in order to see how well the builder performs and we were satisfied with the results we were seeing. As it turned out with the real data the things are different. To my mind this is caused by one (or both or there is something else) of the following:
- the simulation doesn't reflect well the real detector/reconstruction performance;
- the algorithm that calculates the efficiency for real data can not identify fake muons, and gives us inefficiency in cases when we should not reconstruct the fakes.

slava77 commented:
On 1/24/19 10:45 PM, nvoytish wrote:
> @slava77
> in ME11 we saw the biggest inefficiency in comparison with other
> stations, that's why we focused on ME11 (in presentation and while
> developing the solution for the improvement). But the improvement is
> applied for all stations.
> The plots you show are not surprising for me. When we developed the RU
> builder we were using MC samples in order to see how well the builder
> performs and we were satisfied with the results we were seeing. As it
> turned out with the real data the things are different. To my mind this
> is caused by one (or both or there is something else) of the following:
> 
>   * the simulation doesn't reflect well the real detector/reconstruction
>     performance;
>   * the algorithm that calculates the efficiency for real data can not
>     identify fake muons, and gives us inefficiency in cases when we
>     should not reconstruct the fakes.
> 
> 

My tests do not have enough events to measure the efficiency changes
expected from this update.
From your slides I see in ME11 it looks like 0.5% expected or other
chambers in CSC perhaps 0.05-0.1% increase expected.
I was not arguing about the efficiency gain being there.

Still, the question is if an increase of 0.05-0.1% in other chambers is
worth that increase of fake soft muons by ~20%.
Was it confirmed as OK by the muon POG?

slava77 commented:
On 1/25/19 12:22 AM, Slava Krutelyov wrote:
> On 1/24/19 10:45 PM, nvoytish wrote:
>> @slava77
>> in ME11 we saw the biggest inefficiency in comparison with other
>> stations, that's why we focused on ME11 (in presentation and while
>> developing the solution for the improvement). But the improvement is
>> applied for all stations.
>> The plots you show are not surprising for me. When we developed the RU
>> builder we were using MC samples in order to see how well the builder
>> performs and we were satisfied with the results we were seeing. As it
>> turned out with the real data the things are different. To my mind this
>> is caused by one (or both or there is something else) of the following:
>>
>> * the simulation doesn't reflect well the real detector/reconstruction
>> performance;
>> * the algorithm that calculates the efficiency for real data can not
>> identify fake muons, and gives us inefficiency in cases when we
>> should not reconstruct the fakes.
>>
>>
> 
> My tests do not have enough events to measure the efficiency changes
> expected from this update.
>>From your slides I see in ME11 it looks like 0.5% expected or other
> chambers in CSC perhaps 0.05-0.1% increase expected.
> I was not arguing about the efficiency gain being there.
> 
> Still, the question is if an increase of 0.05-0.1% in other chambers is
> worth that increase of fake soft muons by ~20%.
> Was it confirmed as OK by the muon POG?
> 
> —

I should correct myself a bit: the efficiency increases that I was
reading from your slides are at the segment level.
At the level of a muon reconstruction the impact should be much smaller
(likely below 0.1% for all reco muons even in ME11 range and much
smaller than that in other areas with CSC),
which is more consistent with what I see.

Do you happen to have already the estimates from the tag and probe for
the muon reco efficiency gains?

The cost of generating more fakes remains and the gains in muon reco are
still to be quantified more clearly.

ptcox commented:
Hi Slava, everyone,

Let's be clear what we are trying to do here: recover segments that Tag 
& Probe predicts should have been reconstructed but were not. At the 
same time we have found that the old ST segment builder DID reconstruct 
some of these segments, whereas RU did not.

The responsibility of local reconstruction is to reconstruct segments of 
charged particle tracks crossing the CSCs. It has no responsibility for 
muon identification, or even identifying if a charged track originates 
at the primary collision vertex. To some extent, the RU segment builder 
has folded in some loose association with the IP by using roads 
compatible with particles originating somewhere in that location, but we 
still want to be able to reconstruct segments from cosmics and from 
displaced vertices so these constraints are LOOSE.

In particular, I do not think it makes sense to have local 
reconstruction inefficient for ANY real IP muon segments, even if that 
also reduces efficiency for muons and other charged tracks that some 
physics analyses may want to reject. Our task is to make sure we can do 
as well as possible on IP muons.

So far we only have partial results on T&P efficiencies with the old and 
new RU versions. But, again,. I do not think the precise values of those 
are the issue: that is that we want to recover segments which SHOULD 
have been reconstructed, according to Tag & Probe.

Regards, Tim



Slava Krutelyov wrote on 1/25/19 09:42:
> On 1/25/19 12:22 AM, Slava Krutelyov wrote:
> > On 1/24/19 10:45 PM, nvoytish wrote:
> >> @slava77
> >> in ME11 we saw the biggest inefficiency in comparison with other
> >> stations, that's why we focused on ME11 (in presentation and while
> >> developing the solution for the improvement). But the improvement is
> >> applied for all stations.
> >> The plots you show are not surprising for me. When we developed the RU
> >> builder we were using MC samples in order to see how well the builder
> >> performs and we were satisfied with the results we were seeing. As it
> >> turned out with the real data the things are different. To my mind this
> >> is caused by one (or both or there is something else) of the following:
> >>
> >> * the simulation doesn't reflect well the real detector/reconstruction
> >> performance;
> >> * the algorithm that calculates the efficiency for real data can not
> >> identify fake muons, and gives us inefficiency in cases when we
> >> should not reconstruct the fakes.
> >>
> >>
> >
> > My tests do not have enough events to measure the efficiency changes
> > expected from this update.
> >>From your slides I see in ME11 it looks like 0.5% expected or other
> > chambers in CSC perhaps 0.05-0.1% increase expected.
> > I was not arguing about the efficiency gain being there.
> >
> > Still, the question is if an increase of 0.05-0.1% in other chambers is
> > worth that increase of fake soft muons by ~20%.
> > Was it confirmed as OK by the muon POG?
> >
> > —
>
> I should correct myself a bit: the efficiency increases that I was
> reading from your slides are at the segment level.
> At the level of a muon reconstruction the impact should be much smaller
> (likely below 0.1% for all reco muons even in ME11 range and much
> smaller than that in other areas with CSC),
> which is more consistent with what I see.
>
> Do you happen to have already the estimates from the tag and probe for
> the muon reco efficiency gains?
>
> The cost of generating more fakes remains and the gains in muon reco are
> still to be quantified more clearly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457497954>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2Fniwhvi0FiVBqWq8I_XycdIZ_56cUks5vGsN7gaJpZM4aD685>.
>

slava77 commented:
On 1/25/19 2:48 AM, ptcox wrote:
> The responsibility of local reconstruction is to reconstruct segments of
> charged particle tracks crossing the CSCs.

Hi Tim.

I would still assume that you should not reconstruct multiple segments
in a place of one
or reconstruct segments from other BX, or from neutrons, or simply if
the sources can not be known (at least in simulation).
Also, if a segment is found for a particle track,
it should match that track parameters well.

So far, I've only seen that more segments are found per high-pt prompt muon.
This is good.
In case all other features (fake/duplicates and correctness of parameters)
were already studied, please point me to the results/slides.

Thank you.

ptcox commented:
Hi Slava,

Comments in line...

Regards, Tim

Slava Krutelyov wrote on 1/25/19 14:35:
> On 1/25/19 2:48 AM, ptcox wrote:
> > The responsibility of local reconstruction is to reconstruct segments of
> > charged particle tracks crossing the CSCs.
>
> Hi Tim.
>
> I would still assume that you should not reconstruct multiple segments
> in a place of one

I agree - we do not want to spuriously increase segment multiplicity
> or reconstruct segments from other BX,
I disagree - we want to reconstruct everything in the detector
> or from neutrons,
I disagree - if hits from a track originated from a neutron forms a 
segment then we should reconstruct it
> or simply if
> the sources can not be known (at least in simulation).
Not sure what you mean
> Also, if a segment is found for a particle track,
> it should match that track parameters well.
I sort of agree, but I'm not convinced the track reconstruction always 
works 'properly' (there are many cases I've seen where there is clearly 
a segment in the detector but the track reconstruction has dropped it)
>
> So far, I've only seen that more segments are found per high-pt prompt 
> muon.
> This is good.
> In case all other features (fake/duplicates and correctness of parameters)
> were already studied, please point me to the results/slides.
That is not the task of local reconstrcution!
>
> Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457574439>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FnpgMapxkQYZuK4-YLwe0Z26wX-bZks5vGwgmgaJpZM4aD685>.
>

slava77 commented:
>>  In case all other features (fake/duplicates and correctness of parameters)
>> were already studied, please point me to the results/slides.
> That is not the task of local reconstrcution!

Just to make sure I understood you correctly: I read the above statement as the local reconstruction is not responsible for reconstructing segments in the correct position or in a correct direction or for reduction of duplicate or fake segments. Is this really correct?

slava77 commented:
May I rephrase my comments about cost benefit assessment based solely on the local reco objects (earlier arguments using higher level objects were a bit in conflict of responsibilities).

In the reco meeting the updates were advertised as something that increases the matching segment reconstruction by about 0.5% for ME11 and a factor of a few less (close to 0.1%) for all other CSC chambers.

In Z->mumu sample with pileup I see an increase in the number of total reconstructed segments by 21%.
![all_sign1063vsorig_zmm13tevpu2018wf11042p0c_cscdetidcscsegmentsownedrangemap_cscsegments__reco_obj_collection__data__degreesoffreedom](https://user-images.githubusercontent.com/4676718/51752217-c5955b00-206b-11e9-9335-671ce43e9321.png)
A naive guess is that out of the 21% added, only a fraction below 0.5% absolute (or at most 0.5/21\~2\% relative) are real segments useful to increase the matching segment reconstruction.
Please provide some details to justify that the other (as much as 98% of all added segments) are reconstructed as expected.

ptcox commented:
Hi Slava,

I am just trying to say CSC local reco is supposed to reconstruct track 
segments, and that implies direction and position ARE where they should 
be. Reconstructing a segment means *reconstructing* a segment, not 
creating some virtual object far from where the particle passed through 
the chamber,

I do not like the terminology 'fake' segments. If you mean 'spurious' 
segments, that is segments that are not really there then I agree we do 
NOT want to reconstruct them. If you mean 'segments from muons that some 
arbitrary physics analysis does not want to consider' then I believe we 
DO want to reconstruct them. And likewise we should reconstruct segments 
from charged particles that are not muons: if they give hits in the 
detector then there is no general, simple, way we can distinguish them 
from hits from muons.

I do not think what I am saying is controversial. But I am beginning to 
think YOU might believe local reco should be doing more than this?

We don't make any attempt, so far, to reject accidental overlap of hits 
that lead to spurious segment patterns, although we do aim, one day, to 
reduce the time window when building rechits, that will go in that 
direction.

Regards, Tim

Slava Krutelyov wrote on 1/25/19 15:35:
>
>         In case all other features (fake/duplicates and correctness of
>         parameters)
>         were already studied, please point me to the results/slides.
>         That is not the task of local reconstrcution!
>
> Just to make sure I understood you correctly: I read the above 
> statement as the local reconstruction is not responsible for 
> reconstructing segments in the correct position or in a correct 
> direction or for reduction of duplicate or fake segments. Is this 
> really correct?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457591791>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FnirkmfLkLucuNPaiRFdQwRm8B4cuks5vGxZIgaJpZM4aD685>.
>

slava77 commented:
Hi Tim

On 1/25/19 6:52 AM, ptcox wrote:
> I do not like the terminology 'fake' segments. If you mean 'spurious'
> segments, that is segments that are not really there then I agree we do

I meant the segments that are not really there (with perhaps some degree
of fuzziness in the the definition of what "really" is; e.g. IIRC in MC
we reconstruct hits in chambers without a pre-trigger).

> NOT want to reconstruct them. If you mean 'segments from muons that some
> arbitrary physics analysis does not want to consider' then I believe we
> DO want to reconstruct them. And likewise we should reconstruct segments
> from charged particles that are not muons: if they give hits in the
> detector then there is no general, simple, way we can distinguish them
> from hits from muons.
> 
> I do not think what I am saying is controversial. But I am beginning to
> think YOU might believe local reco should be doing more than this?

(mind that this was advertised to me as a tiny change)

well, when suddenly more than 20% of segments appear (for, apparently,
an increase of 0.5% or less in efficiency) questions of responsibility
easily arise in my mind.
Someone will have to deal with this, especially if most of the added
objects will have to be rejected at the analysis stage.
Logically, this is easier done at the source.
That's why I started asking the CSC DPG experts for inputs first.

ptcox commented:
Hi Slava,

Yes, this was advertised to me as a minor change too, so I am a bit 
surprised as well.
However, I am not sure I trust your segment plot - it is not obvious to 
me in that plot what exactly is being shown. What is the x axis? (e.g. 
what are the values 2.5, 4.5, 6.5, 8.5?) Messing with a RangeMap was 
always beyond me. Do you think the overall numbers there make sense?

If it is the case that there is a 21% increase in the number of segments 
then I agree it seems a large increase. But I suppose it might just be 
the way things are and we have to live with it. The segment builder 
wasn't magically tuned to reject segment from pileup, but we DO see 
cases where it is rejecting segments from Z muons that should be built. 
I am unhappy to live with a situation in which we  'accidentally' accept 
such an inefficiency just because that inefficiency also kills pileup 
segments.

Would you recommend that we leave the known inefficiency because it also 
reduces pileup segments? I guess that is an approach that could be 
considered reasonable. I don't like it, but that's just me.

I hope Nikolay and Vladimir will investigate this issue in more detail. 
I don't really know what else to do.

Regards, Tim




Slava Krutelyov wrote on 1/25/19 16:14:
> Hi Tim
>
> On 1/25/19 6:52 AM, ptcox wrote:
> > I do not like the terminology 'fake' segments. If you mean 'spurious'
> > segments, that is segments that are not really there then I agree we do
>
> I meant the segments that are not really there (with perhaps some degree
> of fuzziness in the the definition of what "really" is; e.g. IIRC in MC
> we reconstruct hits in chambers without a pre-trigger).
>
> > NOT want to reconstruct them. If you mean 'segments from muons that some
> > arbitrary physics analysis does not want to consider' then I believe we
> > DO want to reconstruct them. And likewise we should reconstruct segments
> > from charged particles that are not muons: if they give hits in the
> > detector then there is no general, simple, way we can distinguish them
> > from hits from muons.
> >
> > I do not think what I am saying is controversial. But I am beginning to
> > think YOU might believe local reco should be doing more than this?
>
> (mind that this was advertised to me as a tiny change)
>
> well, when suddenly more than 20% of segments appear (for, apparently,
> an increase of 0.5% or less in efficiency) questions of responsibility
> easily arise in my mind.
> Someone will have to deal with this, especially if most of the added
> objects will have to be rejected at the analysis stage.
> Logically, this is easier done at the source.
> That's why I started asking the CSC DPG experts for inputs first.
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457604685>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FntpRdmPDaljrCKNQf0BVlYAlS_wLks5vGx9egaJpZM4aD685>.
>

slava77 commented:
Hi Tim

On 1/25/19 7:32 AM, ptcox wrote:
> Hi Slava,
> 
> Yes, this was advertised to me as a minor change too, so I am a bit
> surprised as well.
> However, I am not sure I trust your segment plot - it is not obvious to
> me in that plot what exactly is being shown. What is the x axis? (e.g.
> what are the values 2.5, 4.5, 6.5, 8.5?) Messing with a RangeMap was
> always beyond me. Do you think the overall numbers there make sense?

the plot shows the degreesOfFreedom variable in the CSCSegments.
Red is with this PR, black is the baseline release (without this PR).

> 
> If it is the case that there is a 21% increase in the number of segments
> then I agree it seems a large increase. But I suppose it might just be
> the way things are and we have to live with it. The segment builder
> wasn't magically tuned to reject segment from pileup, but we DO see
> cases where it is rejecting segments from Z muons that should be built.
> I am unhappy to live with a situation in which we  'accidentally' accept
> such an inefficiency just because that inefficiency also kills pileup
> segments.
> 
> Would you recommend that we leave the known inefficiency because it also
> reduces pileup segments? I guess that is an approach that could be
> considered reasonable. I don't like it, but that's just me.

It would be nice to start with some investigation of the extra segments
(perhaps using ST as a reference, since it is the reference used in the
RU reco updates)

[with a caveat of comparing not exactly the same thing]
when the RU algorithm was introduced in #14718
there was a reduction in the number of segments by 6%
https://cloud.githubusercontent.com/assets/4676718/16587578/35c42b62-4280-11e6-802d-06df0ca3a107.png
here we have an increase by 21%.
So, naively, ST still would produce around 15% fewer segments.

If the extra segments are really just pileup segments well-matching
charged particle tracks crossing the CSC chambers, then this is a good
reason to keep them.
I  think that there is very likely a way to restore the known
inefficiency (at least most of it)
and at the same time keep the number of extra segments not contributing
to this added efficiency at minimum.
I tried to state this with my earliest github comment on this subject;
sorry if that was not clear then.





> 
> I hope Nikolay and Vladimir will investigate this issue in more detail.
> I don't really know what else to do.
> 
> Regards, Tim
> 
> 
>

nvoytish commented:
Hi all!

In order to reduce the number of "unwanted" segments the chi2 filter will not work on its own. This needs a more detailed and profound research. 
By the way, the events that we were looking at with the inefficiency contained the hits that while assembled in segment were giving a quite big chi2 because the trajectory that they describe is far from a straight line.

Best regards,
Nikolay 

________________________________________
From: Slava Krutelyov [notifications@github.com]
Sent: 26 January 2019 06:12
To: cms-sw/cmssw
Cc: Nikolay Voytishin; Mention
Subject: Re: [cms-sw/cmssw] Update RU builder (#25684)

looking at some CSCOffline DQM plots (red is with this PR) in ZMM PU sample in this case, there are some hints of what is happening.

[wf11042_csc_seg_anode_time]<https://user-images.githubusercontent.com/4676718/51782890-2319be00-20e5-11e9-8a18-f959617016bc.png>

central BX counts are up by about 10%, compared to the out-of-time BX increasing by 30%.

[wf11042_csc_seg_nhits]<https://user-images.githubusercontent.com/4676718/51782921-de425700-20e5-11e9-8e8d-5280db185d79.png>

The majority of the added segments have a poor chi2 prob
[wf11042_csc_seg_chi2prob]<https://user-images.githubusercontent.com/4676718/51782905-67a55980-20e5-11e9-8bf8-a39a1accb344.png>
Perhaps this can be used as a simple filter (at least for the segments which are less likely prompt; like the ones that were added with this PR if that's easily separable).

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457803110>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFgBEga-ywbQYjjC8tYPu9XgPZdVZUV7ks5vG-PWgaJpZM4aD685>.

nvoytish commented:
The only reason that I can think of is that the trajectory bends on the scale of a CSC chambers. This means that it's a low pt particle.
Once again as Tim mentioned, if there are hits registered, why not reconstruct segments from them even if these hits are not from muons.
The increase in segment that you showed is ok for me. If you recall the comparison between the ST algo with RU , the ST algo was reconstructing 3point segments on a more than 50% ratio. While here we see a uniform increase in the number of segments of each length. 



________________________________________
From: Slava Krutelyov [notifications@github.com]
Sent: 26 January 2019 18:29
To: cms-sw/cmssw
Cc: Nikolay Voytishin; Mention
Subject: Re: [cms-sw/cmssw] Update RU builder (#25684)

In order to reduce the number of "unwanted" segments the chi2 filter will not work on its own. This needs a more detailed and profound research.
By the way, the events that we were looking at with the inefficiency contained the hits that while assembled in segment were giving a quite big chi2 because the trajectory that they describe is far from a straight line.

If most of the new segments with useful efficiency gain have a bad chi2prob,
perhaps something easy can be done by accepting the bad chi2prob segments in a combination with a selection only in ME11 (and in some other most significantly affected chamber) and/or with strip/wire times only in the central BX.

Is there a good physics reason (guess) for the badness in the chi2? (e.g. the drift model used for the hit reconstruction is expected to deviate in some regions from what is done on the sim/digi side)
This may allow to open up the reconstruction windows only in some regions

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457849316>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFgBEudYFnfz7Q6P9lZpIu_8xbC42fi8ks5vHJCRgaJpZM4aD685>.

slava77 commented:
> What I'd really like to see is the DQM plots for a set of relvals for old and new segment builder. It seems Slava has already seen ZMM+pileup. Can we get our hands on that set of plots too?

I copied the DQM files for ZMM+pileup to eos:
- /store/user/slava77/reltestdata/CMSSW_10_4_0_pre2-orig/11042.0_ZMM_13_DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root for the baseline
- /store/user/slava77/reltestdata/CMSSW_10_4_0_pre2-sign1063/11042.0_ZMM_13_DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root for this PR

ptcox commented:
Thanks for doing this. I've copied them to my Mac and I am exploring 
with root. A bit clunky, because I am not an expert with root (which 
I've just installed on my Mac too). I don't think I'm going to learn 
very much though - it takes too much time. It's more that if we had a 
full set of relvals then one could browse through the DQM and attempt to 
learn things like how SIngleMuon Pt10, 100, 1000, ZMM, BeamHalo, etc. 
change in simulation, and how things change in real SingleMiuon, 
DoubleMuon, and ZeroBias.

Regards, Tim

Slava Krutelyov wrote on 1/28/19 20:22:
>
>     What I'd really like to see is the DQM plots for a set of relvals
>     for old and new segment builder. It seems Slava has already seen
>     ZMM+pileup. Can we get our hands on that set of plots too?
>
> I copied the DQM files for ZMM+pileup to eos:
>
>   * /store/user/slava77/reltestdata/CMSSW_10_4_0_pre2-orig/11042.0_ZMM_13_DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
>     for the baseline
>   * /store/user/slava77/reltestdata/CMSSW_10_4_0_pre2-sign1063/11042.0_ZMM_13_DQM_V0001_R000000001__Global__CMSSW_X_Y_Z__RECO.root
>     for this PR
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-458265816>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FnlB2VAfZ1HBhR8Asf6DJCKBmMcGOks5vH03_gaJpZM4aD685>.
>

slava77 commented:
@nvoytish @ptcox 
please clarify on the status of investigations and possible updates in this PR

ptcox commented:
Hi Slava,

Sorry, overwhelmed with CMS Week last week,  PPD Workshop this week, and 
a Muon Workshop next week.

I presented at the CSC Meeting on 30 January 
https://indico.cern.ch/event/794227/.
The general feeling is that we want the updates, and reconstructing more 
segments in pileup events is good not bad because we want to reconstruct 
everything that looks like a segment.

On the other hand I can understand if you are happier to have the 
unmodified version be a segments-from-pileup filter and prefer to leave 
things unchanged for the legacy reprocessing. If that is your decision, 
then we still want the update included after that. Our job is to detect 
and reconstruct segments from muons, independent of how or where they 
originate.

We could activate the restricted anode-time window that was introduced 
for rechit building a couple of years ago, that will reduce out-of-time 
hits and segments, but it would be cleaner to introduce the two changes 
(RU segment builder updates, reduced anode time window in rechit 
building) in different CMSSW releases so we can clearly distinguish 
which is doing what. (I remind you that we couldn't activate the reduced 
anode time window in the past because the position of BX0 in simulation 
was set to a fixed time bin, which differed from the bin in real data. 
But that position has since been changed in simulation so that it 
matches the bin in real data: from 10_2_0_pre2.)

Regards, Tim

Slava Krutelyov wrote on 2/7/19 15:48:
>
> @nvoytish <https://github.com/nvoytish> @ptcox <https://github.com/ptcox>
> please clarify on the status of investigations and possible updates in 
> this PR
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-461450736>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2Fnsyv-n8yLffu3ABPwkHUVGFi4CDZks5vLDzWgaJpZM4aD685>.
>

slava77 commented:
On 2/12/19 9:41 AM, ptcox wrote:
> Our job is to detect
> and reconstruct segments from muons, independent of how or where they
> originate.

Hi Tim,

Thank you for the clarifications on the intentions/plans/preferences.
If I recall correctly, the majority of the added segments have a bad
chi2prob.
So, they do not look like segments [straight lines] for some reason(s).

did you (CSC, perhaps Nikolay)
check if the extra segments are indeed expected to be there.
- do we generally get just one segment
	in the area of hits that looks like one we should have one
- was there some understanding of the bad chi2prob cases?
	Make sure that hits not related to a charged track like from neutrons
	or from deltas do not contribute.
	Is it natural for segments from OOT PU to have a bad chi2prob?

ptcox commented:
Hi Slava,

I don't know the answers to your questions.

However, the CSC occupancy is still pretty low, even in these pileup 
situations. It is unreasonable that random individual hits in different 
layers of a CSC are getting artificially associated to a segment - there 
just aren't so many random 1-layer or even 2-layer hits like that. 
Basically, there's always a segment, but whether we build it or not can 
depend on the algorithm's parameters. I think the poor chi2 prob/high 
chi2 are more a reflection of underestimated rechit uncertainties than 
that the hits are not 'really' on a segment. (And remember our readout 
requirement of a trigger primitive in coincidence with an L1A before a 
16-strip x 6 layer front-end board is read out. Of course that 
requirement may not be in simulation... but I think it was you who 
argued that was not an issue in simulation.)

It is also the case, I suppose, that the muons in the min bi sample are 
low momentum so perhaps their hits tend to deviate from a straight line 
segment. I think Nikolay was arguing that.

There are certainly lots of things one could look at. What we're driven 
by is the known failures in Tag & Probe. And if to recover those we have 
to build more OOT segments then we have to live with it. Setting up a 
task to investigate if OOT segments have different patterns of hits to 
in-time segments seems like major work that someone might find 
interesting, but I don't think we have the resources to start on that. 
To start, we'd have to reexamine the rechit builder, re-tune it's 
parameters, and the rechit uncertainties, and work until the the segment 
chi2 distributions are thoroughly understood. I can't see that happening 
imminently - there just doesn't seem to be any overwhelming motivation 
to put in that effort.

Regards, Tim



Slava Krutelyov wrote on 2/12/19 19:03:
> On 2/12/19 9:41 AM, ptcox wrote:
> > Our job is to detect
> > and reconstruct segments from muons, independent of how or where they
> > originate.
>
> Hi Tim,
>
> Thank you for the clarifications on the intentions/plans/preferences.
> If I recall correctly, the majority of the added segments have a bad
> chi2prob.
> So, they do not look like segments [straight lines] for some reason(s).
>
> did you (CSC, perhaps Nikolay)
> check if the extra segments are indeed expected to be there.
> - do we generally get just one segment
> in the area of hits that looks like one we should have one
> - was there some understanding of the bad chi2prob cases?
> Make sure that hits not related to a charged track like from neutrons
> or from deltas do not contribute.
> Is it natural for segments from OOT PU to have a bad chi2prob?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-462868193>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2Fnpg73-VffLOdHod7KwBgF_vGhZImks5vMwH7gaJpZM4aD685>.
>

slava77 commented:
> On the other hand I can understand if you are happier to have the unmodified version be a segments-from-pileup filter and prefer to leave things unchanged for the legacy reprocessing. If that is your decision, then we still want the update included after that. Our job is to detect and reconstruct segments from muons, independent of how or where they originate.

I don't think that with or after the UL makes much of a difference.

I have a somewhat practical proposal: please make this change configurable.
We can then request a special relval with just this change.
Once it is validated, the change can be included in the default release setup.

@nvoytish @ptcox please clarify if this looks acceptable to you.
Thank you.

slava77 commented:
@nvoytish @ptcox please comment on 
https://github.com/cms-sw/cmssw/pull/25684#issuecomment-462921631

Thank you.

ptcox commented:
Yes I think that's a good idea. Nikolay and Validmir? Comments?

Tim

Slava Krutelyov wrote on 2/12/19 21:28:
>
>     On the other hand I can understand if you are happier to have the
>     unmodified version be a segments-from-pileup filter and prefer to
>     leave things unchanged for the legacy reprocessing. If that is
>     your decision, then we still want the update included after that.
>     Our job is to detect and reconstruct segments from muons,
>     independent of how or where they originate.
>
> I don't think that with or after the UL makes much of a difference.
>
> I have a somewhat practical proposal: please make this change 
> configurable.
> We can then request a special relval with just this change.
> Once it is validated, the change can be included in the default 
> release setup.
>
> @nvoytish <https://github.com/nvoytish> @ptcox 
> <https://github.com/ptcox> please clarify if this looks acceptable to you.
> Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-462921631>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FnoJNfJj7-dV53kQmX0AYJe1e5vc6ks5vMyQHgaJpZM4aD685>.
>

nvoytish commented:
> > On the other hand I can understand if you are happier to have the unmodified version be a segments-from-pileup filter and prefer to leave things unchanged for the legacy reprocessing. If that is your decision, then we still want the update included after that. Our job is to detect and reconstruct segments from muons, independent of how or where they originate.
> 
> I don't think that with or after the UL makes much of a difference.
> 
> I have a somewhat practical proposal: please make this change configurable.
> We can then request a special relval with just this change.
> Once it is validated, the change can be included in the default release setup.
> 
> @nvoytish @ptcox please clarify if this looks acceptable to you.
> Thank you.

Hi all!

What's the difference? You'll see similar comparison results. Or I am missing something?
It is not a problem to make the change configurable(as a temporary feature, because it will look ugly in the code).

Best,
Nikolay

slava77 commented:
On 2/19/19 3:00 PM, nvoytish wrote:
> What's the difference? You'll see similar comparison results. Or I am
> missing something?
> It is not a problem to make the change configurable(as a temporary
> feature, because it will look ugly in the code).
> 

My proposal was to do the following:
1. make the change configurable
2. keep the default corresponding to the current baseline behavior in
the release (no differences in physics)
3. a special set of relvals is requested with the configuration which
has new proposed physics behavior
4. After the validation of 3. is confirmed, there is a small pull
request to enable the new behavior (either at the config level or by
switching over to the current PR state where the change is hardcoded.

slava77 commented:
@nvoytish @ptcox 
please clarify on the status of expected updates in this PR.
Thank you.

nvoytish commented:
I'll be preparing the optional configuration this week. Unfortunately in the last couple of weeks there was an overload of tasks at our side.

nvoytish committed:
Optional enlargement of the thresholds added

The optional enlargement of the threshold added in order to test initial changes proposed in this PR. The bool enlarge is used as a flag. If false the segment builder behaves as without any changes proposed in this PR.

cmsbuild commented:
The code-checks are being triggered in jenkins.

nvoytish committed:
Optional enlargement of the thresholds added

the bool flag "enlarge" added and set to false.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25684/8548

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #25684 was updated. @cmsbuild, @perrotta, @slava77 can you please check and sign again.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

boudoul commented:
Hi @nvoytish , could you please edit the title to make it a bit more explicit , at least mentioning that this is for CSC , maybe 'Update CSC  Segment Builder'  for example ? Thanks !

cmsbuild commented:
+1
Tested at: 995a4cf31ac53de75d7105e0dca076d1a4d3d101
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/33313/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/33313/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3098286
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3098088
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

slava77 commented on RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc:
```diff
@@ -391,11 +414,10 @@ bool CSCSegAlgoRU::areHitsCloseInR(const AlgoState& aState, const CSCRecHit2D* h
+  if(enlarge) {
+    return (gp2.perp() > ((gp1.perp() - aState.dRMax*aState.strip_iadd*maxWG_width[iStn])*h2z)/h1z && gp2.perp() < ((gp1.perp() + aState.dRMax*aState.strip_iadd*maxWG_width[iStn])*h2z)/h1z)? true:false;
+  }else{
+    return (gp2.perp() > ((gp1.perp() - aState.dRMax*maxWG_width[iStn])*h2z)/h1z && gp2.perp() < ((gp1.perp() + aState.dRMax*maxWG_width[iStn])*h2z)/h1z)? true:false;
```
please add some newlines to improve readability

slava77 replied:
> please add some newlines to improve readability

sorry for being unclear, I meant to ask for line breaks in these long lines of code. Simple addition of empty lines around them is not really improving the code readability.

slava77 commented on RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc:
```diff
@@ -478,7 +500,11 @@ bool CSCSegAlgoRU::isHitNearSegment(const AlgoState& aState, const CSCRecHit2D*
+  if(enlarge) {
+    return (fabs(phidif) < aState.dPhiIntMax*aState.strip_iadd*pos_str+dphi_incr && fabs(dr) < aState.dRIntMax*aState.strip_iadd*maxWG_width[iStn])? true:false;
+  }else{
+    return (fabs(phidif) < aState.dPhiIntMax*aState.strip_iadd*pos_str+dphi_incr && fabs(dr) < aState.dRIntMax*maxWG_width[iStn])? true:false;
```
please add some newlines to improve readability

slava77 commented on RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.h:
```diff
@@ -162,7 +162,7 @@ class CSCSegAlgoRU : public CSCSegmentAlgorithm {
     int minLayersApart;
     bool debugInfo;
-
+    bool enlarge = false;
```
this should be a configurable parameter.
Please propagate an appropriate logic to set it via the CSCSegmentProducer configuration

nvoytish replied:
Hi Slava!

Do you mean to add it as a parameter to the python config file particularly to RecoLocalMuon/CSCSegment/python/CSCSegmentAlgorithmRU_cfi.py ?

slava77 replied:
right.
The goal is to allow this flag to be changed at configuration step so that some special relval production can be made.

nvoytish replied:
Ok. I am on it)

slava77 commented on RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc:
```diff
@@ -257,12 +272,20 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const CSCChamber* aChamber,
+        aState.chi2Max = 0.25*aState.chi2Max;
+      } else {
+        aState.dPhiMax = 0.5*aState.dPhiMax;
+        aState.dRIntMax = 0.5*aState.dRIntMax;
```
please do not copy-paste essentially the same lines of code.
Define a scale constant (1. for enlarge=false; and 2. for enlarge = true)

slava77 commented on RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc:
```diff
@@ -128,13 +133,23 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const CSCChamber* aChamber,
+        aState.chi2Max = 4*aState.chi2Max;
+      } else {
+        aState.dRMax = 2.0;
+	aState.dPhiMax = 2*aState.dPhiMax;
```
please do not copy-paste essentially the same lines of code.
Define a scale constant (1. for enlarge=false; and 2. for enlarge = true)

slava77 reviewed (commented):
the most important change missing in this PR now is to make the ```enlarge``` parameter configurable

nvoytish committed:
making the bool parameter "enlarge" configurable

making the bool parameter "enlarge" configurable + some code cleaning

nvoytish committed:
making the bool parameter "enlarge" configurable

cmsbuild commented:
The code-checks are being triggered in jenkins.

nvoytish committed:
making the bool parameter "enlarge" configurable

making the bool parameter "enlarge" configurable. Set to "False" as default value.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25684/8570

- This PR adds an extra 32KB to repository

cmsbuild commented:
Pull request #25684 was updated. @cmsbuild, @perrotta, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 13920affba9b8ec56ac3b1926225c475168d4ecb
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/33393/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25684/33393/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3114826
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3114628
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

slava77 commented:
@ptcox 
Not quite related to this PR as is; I notice that CSCSegAlgoRU is not used in HLT (CSCSegAlgoST is used).
Is there a good reason for this?

ptcox commented:
That's pretty amusing! We certainly didn't intend that! CMSSW is SUCH a 
mess! I wonder what other surprises lie in wait to be uncovered?

Tim

Slava Krutelyov wrote on 3/5/19 18:35:
>
> @ptcox <https://github.com/ptcox>
> Not quite related to this PR as is; I notice that CSCSegAlgoRU is not 
> used in HLT (CSCSegAlgoST is used).
> Is there a good reason for this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-469776587>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2Fnt8sEjMJ4lgTJo-bOmzR-3392eVTks5vTqrSgaJpZM4aD685>.
>

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/25684 	13920af
- code changes are in line with the PR description and the follow up review. 
- jenkins tests pass and comparisons with the baseline show no differences
    - a summary of changes observed with the new tuning of the algorithm enabled can be seen in https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457421185 through https://github.com/cms-sw/cmssw/pull/25684#issuecomment-457803110 

@fabiocos as discussed with @ptcox and @nvoytish  earlier in this PR, the plan is to use the configuration options introduced in this PR to request a special set of relvals produced with the new algorithm tuning enabled. These will be used for an incremental validation of the changes in the CSC segment algorithm. The changes can be enabled with
```--customise_command "for p in process.cscSegments.algo_psets[4].algo_psets: p.enlarge = True\n"``` to be applied to the reco step (step3) of the workflows.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

nvoytish commented:
> @ptcox
> Not quite related to this PR as is; I notice that CSCSegAlgoRU is not used in HLT (CSCSegAlgoST is used).
> Is there a good reason for this?

 As far as I remember the people from the  MuonPOG  were concerned that RU might be too much time consuming. Time is crucial for HLT if I am not mistaken.  
If needed, we can try to make a more profound investigation for this?

ptcox commented:
We have never wanted HLT to differ from Offline in any way at all.

Tim

nvoytish wrote on 3/5/19 19:16:
>
>     @ptcox <https://github.com/ptcox>
>     Not quite related to this PR as is; I notice that CSCSegAlgoRU is
>     not used in HLT (CSCSegAlgoST is used).
>     Is there a good reason for this?
>
> As far as I remember the people from the MuonPOG were concerned that 
> RU might be too much time consuming. Time is crucial for HLT if I am 
> not mistaken.
> If needed, we can try to make a more profound investigation for this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/25684#issuecomment-469792187>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AE2FntXxvOhLV5OiyTF3Hd9xaKLf7KrUks5vTrR6gaJpZM4aD685>.
>

fabiocos commented:
@slava77 thank you for pointing me to https://github.com/cms-sw/cmssw/pull/25684#issuecomment-469787968 , @prebello @zhenhu should take note in view of the RelVal request

fabiocos commented:
+1

the update looks by default off, no effect expected
